File: 2017/AnimatedVectorDrawables/AnimateWriting/app/src/main/java/me/mehdi/animatewriting/MainActivity.java
package me.mehdi.animatewriting;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.drawable.AnimatedVectorDrawable;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {

    ImageView mImage;
    Button mButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mButton = findViewById(R.id.start);
        mImage = findViewById(R.id.imageView);

        mButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ((AnimatedVectorDrawable)mImage.getDrawable()).start();
            }
        });
    }
}

File: 2022/MyVectorGraphics/app/src/main/java/com/codewithmehdi/myvectorgraphics/MainActivity.kt
package com.codewithmehdi.myvectorgraphics

import android.graphics.drawable.AnimatedVectorDrawable
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.ViewGroup
import android.widget.ImageView
import androidx.core.view.forEach
import androidx.vectordrawable.graphics.drawable.AnimatedVectorDrawableCompat
import com.codewithmehdi.myvectorgraphics.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)

        setContentView(binding.root)

        binding.startAVDsStatic.setOnClickListener {
            binding.AVDsContainer.forEach {view->
                if (view is ImageView){
                    (view.drawable as AnimatedVectorDrawable).start()
                }
            }
        }


        binding.startAVDDynamic.setOnClickListener{
            val avd = AnimatedVectorDrawableCompat.create(this, R.drawable.avd_progress_circle)
            findViewById<ViewGroup>(android.R.id.content).addView(
                ImageView(this).apply{
                    setImageDrawable(avd)
                }
            )

            avd?.start()
        }

        binding.startAVDBackground.setOnClickListener{button->
            button.setBackgroundResource(R.drawable.avd_writing_tick)
            (button.background as AnimatedVectorDrawable).start()
        }



    }
}